src/oop_ext/interface/_tests/test_interface.py

```diff
@@ -1077,3 +1077,39 @@ def AfterCaption(*args):
 
     assert Foo.GetCaption() == "Foo"
     assert Foo().GetValues("m") == [0.1, 10.0]
+
+
+def testGenericInterface() -> None:
+    """Generic interfaces need to support checking on both generic and non-generic implementers"""
+    from typing import FrozenSet, Generic, TypeVar
+
+    T = TypeVar("T", covariant=True)
+
+    class IFoo(Interface, Generic[T], TypeCheckingSupport):
+        # Class instance defined here as a workaround for this class to work in Python 3.6.
+        __abstractmethods__: FrozenSet = frozenset()
+
+        def GetOutput(self) -> T:  # type:ignore[empty-body]
+            ...
```

Comment:
To solve this new warning we are decorating interface methods with `@abstractmethod`, so mypy won't complain. Also let's make the example more comprehensive by also having an input parameter:

```suggestion
        @abstractmethod
        def GetOutput(self, v: T) -> T: 
            ...
```